test/e2e-tests.sh

```diff
@@ -23,136 +23,156 @@ source $(dirname $0)/e2e-common.sh
-for file in ${pipelineRunFiles[@]};do
-  json_curl_envsubst_resource "${file}" "POST" "${TEKTON_PROXY_URL}/pipelineruns" || fail_test "Failed to create pipelinerun: ${file}"
-done
-
-print_diagnostic_info
-# Wait for deployment
-echo "About to check the deployment..."
-deploymentExist=false
-for i in {1..30}
-do
-  wait=$(kubectl wait --namespace tekton-pipelines --for=condition=available deployments/go-hello-world --timeout=30s)
-  echo "WAIT RESULT: $wait"
-  if [ "$wait" = "deployment.apps/go-hello-world condition met" ]; then
-    deploymentExist=true
-    break
-  elif [ "$wait" = "deployment.extensions/go-hello-world condition met" ]; then
-    deploymentExist=true
-    break
-  else
-    echo "About to sleep, getting and describing pod info"
+
+  # API/resource configuration
+  export APP_SERVICE_ACCOUNT="e2e-tests"
+  export PIPELINE_NAME="simple-pipeline-insecure"
+  export IMAGE_RESOURCE_NAME="docker-image"
+  export GIT_RESOURCE_NAME="git-source"
+  export GIT_COMMIT="master"
+  export REPO_NAME="go-hello-world"
+  export REPO_URL="https://github.com/ncskier/go-hello-world"
+  export EXPECTED_RETURN_VALUE="Hello World!"
+  export REGISTRY="gcr.io/${E2E_PROJECT_ID}/${E2E_BASE_NAME}-e2e-img"
+  export TEKTON_PROXY_URL="http://localhost:9097/proxy/apis/tekton.dev/v1alpha1/namespaces/tekton-pipelines"
+  export CSRF_HEADERS_STORE="csrf_headers.txt"
+
+  # Kubectl static resources
+  kubectl apply -f ${tekton_repo_dir}/test/resources/static
+
+  curl -D $CSRF_HEADERS_STORE http://localhost:9097/v1/token
+  export CSRF_TOKEN=`grep -i 'X-CSRF-Token' $CSRF_HEADERS_STORE | sed -e 's/^X-CSRF-Token: //i;s/\r//'`
+  export CSRF_COOKIE=`grep -i 'Set-Cookie' $CSRF_HEADERS_STORE | sed -e 's/Set-Cookie: //i;s/; .*//;s/\r//'`
+
+  # # Create envsubst resources through dashboard proxy
+  # pipelineResourceFiles=($(find ${tekton_repo_dir}/test/resources/envsubst -iname "pipelineresource*.y?ml"))
+  # for file in ${pipelineResourceFiles[@]};do
+  #   json_curl_envsubst_resource "${file}" "POST" "${TEKTON_PROXY_URL}/pipelineresources" || fail_test "Failed to create pipelineresource: ${file}"
+  # done
+
+  # pipelineRunFiles=($(find ${tekton_repo_dir}/test/resources/envsubst -iname "pipelinerun*.y?ml"))
+  # for file in ${pipelineRunFiles[@]};do
+  #   json_curl_envsubst_resource "${file}" "POST" "${TEKTON_PROXY_URL}/pipelineruns" || fail_test "Failed to create pipelinerun: ${file}"
+  # done
+
+  # Kubectl envsubst resources
+  pipelineResourceFiles=($(find ${tekton_repo_dir}/test/resources/envsubst -iname "pipelineresource*.y?ml"))
+  for file in ${pipelineResourceFiles[@]};do
+    cat "${file}" | envsubst | kubectl apply -f - || fail_test "Failed to create pipelineresource: ${file}"
+  done
+
+  pipelineRunFiles=($(find ${tekton_repo_dir}/test/resources/envsubst -iname "pipelinerun*.y?ml"))
+  for file in ${pipelineRunFiles[@]};do
+    cat "${file}" | envsubst | kubectl apply -f - || fail_test "Failed to create pipelinerun: ${file}"
+  done
```

Comment:
One concern i have is that it's not using the dashboard proxy to create the resources.

Reply:
Hmm, this is a big drawback though surely - we're just doing a normal kube apply instead of hitting our PIpelineRuns/PipelineResources APIs which we were previously doing - have I missed something?

Reply:
When testing the read only overlay, calling the proxy won't work (the dashboard SA cannot create resources).
Kubectl will work because it's using the test runner kube context.

Should i use the proxy in read write mode and kubectl in read only mode ?

Reply:
Yep, sounds good to me

Reply:
done